app/src/main/java/org/mozilla/fenix/wallpapers/WallpaperManager.kt

```diff
@@ -30,6 +33,44 @@ class WallpaperManager(private val settings: Settings) {
             wallpaperContainer.setBackgroundResource(newWallpaper.drawable)
         }
         currentWallpaper = newWallpaper
+
+        adjustTheme(wallpaperContainer.context)
+    }
+
+    private fun adjustTheme(context: Context) {
+        val mode = if (currentWallpaper != Wallpaper.NONE) {
+            if (currentWallpaper.isDark) {
+                setDarkThemePreference()
+                AppCompatDelegate.MODE_NIGHT_YES
+            } else {
+                setLightThemePreference()
+                AppCompatDelegate.MODE_NIGHT_NO
+            }
+        } else {
+            setFollowDeviceThemePreference()
+            AppCompatDelegate.MODE_NIGHT_FOLLOW_SYSTEM
+        }
+
+        AppCompatDelegate.setDefaultNightMode(mode)
+        context.asActivity()?.recreate()
+    }
+
+    private fun setDarkThemePreference() {
+        settings.shouldUseDarkTheme = true
+        settings.shouldUseLightTheme = false
+        settings.shouldFollowDeviceTheme = false
```

Comment:
Can we not just replace all of this with a single setting that uses a String enum for the tri-state?

Reply:
Okay, I am now see this was already something existing.